### ISSUE TITLE
🚀 Feature: Scaffolder - clustering of actions

### ISSUE THREAD

**Issue by waldi** [backstage/backstage#23949]
### 🔖 Feature description

It would be nice to be able to cluster actions. 

Example of current implementation:

```yaml
steps:
  - id: fooOne
    action: foo-action

  - id: barOne
    action: bar-action

  - id: barTwo
    action: bar-action

  - id: barThree
    action: bar-action

  - id: fooBarOne
    action: foo-bar-action
```

Example of preferred implementation: 

```
jobs:
  - title: Foo Job
    steps:
      - id: fooOne
        action: foo-action

  - title: Bar Job
    steps:
      - id: barOne
        action: bar-action

      - id: barTwo
        action: bar-action

      - id: barThree
        action: bar-action

  - title: Foo Bar Job
    steps:
      - id: fooBarOne
        action: foo-bar-action
```

Both options of implementing the template should be possible to be backwards compatible.

### 🎤 Context

In my use-case, we write templates with a lot of small atomic actions that could be grouped. This could partially be a solution for large action lists. Instead of showing all the actions we could display the job titles only. 

Related issue:
https://github.com/backstage/backstage/issues/23948

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by awanlin** [comment]
This is an interesting idea, no idea how easy or hard this would be to implement but I personally like it 😎

**Comment by acierto** [comment]
Still would be nice to have a chose how to run it - in parallel or sequentially. 
If to do 3 unrelated to each other tasks, make sense to run them in parallel.
But if the cluster of actions depends on a result of previous cluster of actions, better to keep it sequential.

In our case I combined few actions into one, so that user doesn't see some very technical step titles. 
I see the value of supporting these feature too.

**Comment by waldi** [comment]
Azure Pipelines have the possibility to specify the `parallel` strategy:

```
- job: string  # name of the job, A-Z, a-z, 0-9, and underscore
  displayName: string  # friendly name to display in the UI
  dependsOn: string | [ string ]
  condition: string
  strategy:
    parallel: # parallel strategy
    matrix: # matrix strategy
    maxParallel: number # maximum number simultaneous matrix legs to run
    # note: `parallel` and `matrix` are mutually exclusive
    # you may specify one or the other; including both is an error
    # `maxParallel` is only valid with `matrix`
  continueOnError: boolean  # 'true' if future jobs should run even if this job fails; defaults to 'false'
  pool: pool # agent pool
  workspace:
    clean: outputs | resources | all # what to clean up before the job runs
  container: containerReference # container to run this job inside
  timeoutInMinutes: number # how long to run the job before automatically cancelling
  cancelTimeoutInMinutes: number # how much time to give 'run always even if cancelled tasks' before killing them
  variables: { string: string } | [ variable | variableReference ] 
  steps: [ script | bash | pwsh | powershell | checkout | task | templateReference ]
  services: { string: string | container } # container resources to run as a service container
  uses: # Any resources (repos or pools) required by this job that are not already referenced
    repositories: [ string ] # Repository references to Azure Git repositories
    pools: [ string ] # Pool names, typically when using a matrix strategy for the job
```

https://learn.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml

**Comment by nia-potato** [comment]
this seems really useful, would love to see it merged !

**Comment by acierto** [comment]
The parallel execution of steps will affect UI as well, so it has to be considered as well.

@benjdlambert what do you think about the entire idea of this?

**Comment by benjdlambert** [comment]
Some of the UI work is being discussed in this issue: https://github.com/backstage/backstage/issues/23948
